### DISCUSSION TITLE
Rendering audio files similar to Audacity/Python

### DISCUSSION THREAD

**Discussion by vranicf** [katspaugh/wavesurfer.js#3216] [Q&A] [answered]
Hi,

Is there a way to render audio in a similar fashion to Audacity or Python, without filling up the area below the graph? Is it possible only with a custom drawer?

**Python**
<img width="929" alt="matplotlib" src="https://github.com/katspaugh/wavesurfer.js/assets/122837574/95ad674e-ed85-404c-9360-16773bdd8cb3">

**Wavesurfer**
<img width="653" alt="wavesurfer" src="https://github.com/katspaugh/wavesurfer.js/assets/122837574/7b7052f6-5a9c-4801-a515-66965c768fff">

WaveSurfer is instantiated like this:

```typescript
const wavesurfer = WaveSurfer.create({
        container: wavesurferRef.current,
        waveColor: theme.palette.primary.main,
        cursorColor: "#f44336",
        cursorWidth: 2,
        progressColor: "#6c727f",
        interact: true,
        autoplay: true,
        barHeight: 2,
        normalize: true,
        media: audioPlayerHTMLAudioElementRef.current,
        plugins: [
          Timeline.create({
            container: wavesurferTimelineRef.current,
          }),
        ],
      })
```

Thank you in advance!

**Comment by katspaugh** [comment, marked as answer]
Yes, it's possible only via a [custom `renderFunction` option](https://wavesurfer-js.org/examples/?custom-render.js).
It would be almost the same as the [default render function](https://github.com/katspaugh/wavesurfer.js/blob/main/src/renderer.ts#L312-L354) but with `ctx.stroke()` instead of `ctx.fill()`.

**Comment by vranicf** [comment]
Thanks! I already tried the custom renderer you provided, but it didn't change the rendered figure. I must be missing something.

**Reply by katspaugh** [reply]
Try something like this: https://wavesurfer-js.org/examples/?a6fc58e1a1516f7b89b338c1cfe6e6f3

**Reply by vranicf** [reply]
Thanks! It must be something with the canvas that's preventing me from drawing. It's like renderer function is completely ignored.

**Reply by katspaugh** [reply]
It's time to open the debugger then...
